lib/translator/inquiry.cc

```diff
@@ -0,0 +1,292 @@
+      .protect = (identify_namespace_data.dps.pit == 0 &&
+                  identify_namespace_data.dps.md_start == 0)
+                     ? 0
+                     : 1,
+
+      .cmdque = 1};
+
+  // Shall be set to “NVMe” followed by 4 spaces: “NVMe    “
+  // Vendor Identification is not null terminated.
+  memcpy(result.vendor_identification, kNvmeVendorIdentification.data(),
+         kNvmeVendorIdentification.size());
+
+  // Shall be set to the first 16 bytes of the Model Number (MN) field within
+  // the Identify Controller Data Structure
+  memcpy(result.product_identification, identify_controller_data.mn,
+         sizeof(result.product_identification));
+
+  /*
+  Shall be set to the last 4 ASCII graphic characters in the range of 21h-7Eh
+  (i.e. last 4 non-space characters) of the Firmware Revision (FR) field within
+  the Identify Controller Data Structure.
+  */
+  int idx = 3;
+  for (int i = 7; i >= 0; i--) {
+    if (identify_controller_data.fr[i] != ' ' &&
+        identify_controller_data.fr[i] >= 0x21 &&
+        identify_controller_data.fr[i] <= 0x7e) {
+      result.product_revision_level[idx] = identify_controller_data.fr[i];
+      if (idx-- == 0) break;
+    }
+  }
+
+  if (idx >= 0) {
+    DebugLog("less than four characters set\n");
+  }
+  return result;
+}
+
+scsi_defs::InquiryData BuildStandardInquiry() {
+  // Identify controller results
+  nvme_defs::IdentifyControllerData identify_controller_data;
+  // Identify namespace results
+  nvme_defs::IdentifyNamespace identify_namespace_data;
+  return TranslateStandardInquiryResponse(identify_controller_data,
+                                          identify_namespace_data);
+}
+
+scsi_defs::SupportedVitalProductData BuildSupportedVpdPages() {
+  // TODO: write this after SupportedVitalProductData in the return buffer when
+  // we agree on how to return SCSI responses
+  scsi_defs::PageCode supported_page_list[7] = {
+      scsi_defs::PageCode::kSupportedVpd,
+      scsi_defs::PageCode::kUnitSerialNumber,
+      scsi_defs::PageCode::kDeviceIdentification,
+      scsi_defs::PageCode::kExtended,
+      scsi_defs::PageCode::kBlockLimitsVpd,
+      scsi_defs::PageCode::kBlockDeviceCharacteristicsVpd,
+      scsi_defs::PageCode::kLogicalBlockProvisioningVpd};
+
+  return scsi_defs::SupportedVitalProductData{
+      // Shall be set to 5 indicating number of items supported Vpd pages list
+      // requires. NOTE: document says to set this to 5 but there are 7
+      // entries....
+      .page_length = 5,
+  };
+}
+
+scsi_defs::UnitSerialNumber TranslateUnitSerialNumberVpdResponse(
+    const nvme_defs::IdentifyNamespace &identify_namespace_data) {
+  scsi_defs::UnitSerialNumber result = scsi_defs::UnitSerialNumber();
+  result.peripheral_qualifier =
+      scsi_defs::PeripheralQualifier::kPeripheralDeviceConnected;
+  result.peripheral_device_type =
+      scsi_defs::PeripheralDeviceType::kDirectAccessBlock;
+  result.page_code = 0x80;
```

Comment:
For each possible page code value here, define a named constant in the header file. You can se this constant to test your code instead of comparing to the magic number in your test file. It is discouraged to have magic constants like `0x80` here in your code, since it's not possible to tell what the number means by itself.

Try
```
uint8_t kConstantName = 0x80;
```

Reply:
I will use the PageCode enum we defined